tini/src/seccomp_fd_notify.c

```diff
@@ -179,23 +184,60 @@ struct sock_filter net_filter[] = {
 		}
 	}
 
+	for (int i = 0;
+	     i < (unsigned short)(sizeof(check_filter) / sizeof(check_filter[0])) &&
```

Comment:
Can you add a comment to explain what this code does? 
I don't understand it.

I think I don't understand it because you end up using the `check_filter` variable in 2 ways.
1. It stores the kind of first bpf filter  we need
2. It grows to include the final computed filter with all the individual filters combined.

Maybe have two different variables?

Reply:
We have a variable `filter` that is an array of size `BPF_MAXINSNS + 1`.

We are populating this array upto a max size of `BPF_MAXINSNS` with individual arrays of filters i.e. `check_filter`, `net_filter`, `perf_filter`, `traffic_steering_filter` and `allow_filter` depending on what feature flag is enabled.

The `filter` variable is finally installed using the call to seccomp()